pkg/kn/commands/source/binding/binding.go

```diff
@@ -0,0 +1,121 @@
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/tools/clientcmd"
+	"knative.dev/eventing/pkg/client/clientset/versioned/typed/sources/v1alpha1"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	duckv1beta1 "knative.dev/pkg/apis/duck/v1beta1"
+	"knative.dev/pkg/tracker"
+
+	"knative.dev/client/pkg/kn/commands"
+	sources_v1alpha1 "knative.dev/client/pkg/sources/v1alpha1"
+)
+
+// NewBindingCommand is the root command for all binding related commands
+func NewBindingCommand(p *commands.KnParams) *cobra.Command {
+	bindingCmd := &cobra.Command{
+		Use:   "binding",
+		Short: "Sink binding command group",
+	}
+	bindingCmd.AddCommand(NewBindingCreateCommand(p))
+	bindingCmd.AddCommand(NewBindingUpdateCommand(p))
+	bindingCmd.AddCommand(NewBindingDeleteCommand(p))
+	bindingCmd.AddCommand(NewBindingListCommand(p))
+	bindingCmd.AddCommand(NewBindingDescribeCommand(p))
+	return bindingCmd
+}
+
+var sinkBindingClientFactory func(config clientcmd.ClientConfig, namespace string) (sources_v1alpha1.KnSinkBindingClient, error)
+
+func newSinkBindingClient(p *commands.KnParams, cmd *cobra.Command) (sources_v1alpha1.KnSinkBindingClient, error) {
+	namespace, err := p.GetNamespace(cmd)
+	if err != nil {
+		return nil, err
+	}
+
+	if sinkBindingClientFactory != nil {
+		config, err := p.GetClientConfig()
+		if err != nil {
+			return nil, err
+		}
+		return sinkBindingClientFactory(config, namespace)
+	}
+
+	clientConfig, err := p.RestConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := v1alpha1.NewForConfig(clientConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	return sources_v1alpha1.NewKnSourcesClient(client, namespace).SinkBindingClient(), nil
+}
+
+// Temporary conversions function until we move to duckv1
+func toDuckV1(destination *duckv1beta1.Destination) *duckv1.Destination {
+	return &duckv1.Destination{
+		Ref: destination.Ref,
+		URI: destination.URI,
+	}
+}
+
+func toReference(subjectArg string) (*tracker.Reference, error) {
+	parts := strings.SplitN(subjectArg, ":", 3)
+	if len(parts) < 3 {
+		return nil, fmt.Errorf("invalid subject argument '%s': not in format kind:api/version:nameOrSelector", subjectArg)
+	}
+	kind := parts[0]
+	gv, err := schema.ParseGroupVersion(parts[1])
+	if err != nil {
+		return nil, err
+	}
+	reference := &tracker.Reference{
+		APIVersion: gv.String(),
+		Kind:       kind,
+	}
+	if !strings.Contains(parts[2], "=") {
+		reference.Name = parts[2]
+	} else {
+		selector, err := parseSelector(parts[2])
+		if err != nil {
+			return nil, err
+		}
+		reference.Selector = &metav1.LabelSelector{MatchLabels: selector}
+	}
+	return reference, nil
+}
+
+func parseSelector(labelSelector string) (map[string]string, error) {
+	selector := map[string]string{}
+	for _, p := range strings.Split(labelSelector, ",") {
+		keyValue := strings.SplitN(p, "=", 2)
+		if len(keyValue) != 2 {
+			return nil, fmt.Errorf("invalid subject label selector '%s'. format: key1=value,key2=value", labelSelector)
+		}
```

Comment:
same comment as above, lets use `Split` here instead of `SplitN`

Reply:
Again my argument that with SplitN you can allow labels holding `=` as values. Not sure how realistic this is, but definitely possible. If you use `Split` you have to check the case when its _more than 2_ parts. With SplitN you can be sure that its _at most 2_ parts (but can be less)

Reply:
sounds good, thanks!